robottelo/cli/hammer.py

```diff
@@ -6,17 +7,18 @@
 
 def parse_csv(output):
     """Parse CSV output from Hammer CLI and convert it to python dictionary."""
-    result = []
+    reader = csv.reader(output)
     # Generate the key names, spaces will be converted to dashes "-"
     keys = [
-        header.replace(' ', '-').lower() for header in output.pop(0).split(',')
+        header.replace(' ', '-').lower() for header in reader.next()
     ]
 
     # For each entry, split the values and create a dict mapping each key with
     # each value
-    for line in output:
-        if len(line) > 0:
-            result.append(dict(izip(keys, line.split(','))))
+    result = []
+    for values in reader:
+        if len(values) > 0:
+            result.append(dict(izip(keys, values)))
 
     return result
```

Comment:
Given that we're already refactoring code, how about turning this into a list comprehension?

```python
return  [dict(izip(keys, values)) for values in reader if len(values) > 0]
```